crypto3/libs/multiprecision/include/nil/crypto3/multiprecision/cpp_int_modular/misc.hpp

```diff
@@ -147,7 +147,9 @@ namespace boost {
                         if (a.limbs()[i] != 0)
                             return i * cpp_int_modular_backend<Bits>::limb_bits + 
                                 boost::multiprecision::detail::find_msb(a.limbs()[i]);
-                    } 
+                    }
+                    if (a.limbs()[0] == 0)
```

Comment:
@tigran-at-nil Could you please comment on this?

Reply:
I checked that if the argument is 0,  boost::multiprecision::detail::find_msb() returns value of "-1", which, when casted to an unsigned type, results in maximal value of it (because of underflow). So if the underlying type is 32 bit, it will be one value (2^32-1), if it is 64-bit, the value will be different.

From the other side, the latest comment makes our function "eval_msb_imp()" to return 0 for 2 cases - when input argument is 0 and when it is 1. So it might be confusing...

However, for now it is OK for our code, as we don't run "eval_msb_imp()" for argument of "0". Concluding, I will suggest to add just one more "assert()" at the beginning of "eval_msb_imp()", to check that it is not called with "0" argument.

Reply:
according to sanitizers, find_msb(0) is undefined behavior. I will create a separate PR for this then

Reply:
And we have a check in the line 147:
```
if (a.limbs()[i] != 0)
```

Reply:
moved to a separate PR
https://github.com/NilFoundation/placeholder/pull/45